src/game/phase/phase.c

```diff
@@ -25,23 +25,28 @@ static bool m_Running = false;
 static PHASE m_PhaseToSet = PHASE_NULL;
 static void *m_PhaseToSetArg = NULL;
 
-static GAMEFLOW_OPTION Phase_Control(int32_t nframes);
+static GAMEFLOW_INSTRUCTION Phase_Control(int32_t nframes);
 static void Phase_Draw(void);
 static int32_t Phase_Wait(void);
 static void Phase_SetUnconditionally(const PHASE phase, void *arg);
 
-static GAMEFLOW_OPTION Phase_Control(int32_t nframes)
+static GAMEFLOW_INSTRUCTION Phase_Control(int32_t nframes)
 {
     if (g_GameInfo.override_option != GF_PHASE_CONTINUE) {
-        const GAMEFLOW_OPTION result = g_GameInfo.override_option;
         g_GameInfo.override_option = GF_PHASE_CONTINUE;
-        return result;
+        return (GAMEFLOW_INSTRUCTION) {
+            .instruction = g_GameInfo.override_option,
+            .param = 0,
+        };
```

Comment:
Oh no! You return `GF_PHASE_CONTINUE` here but the idea is to use the value prior to setting it!